frontend/src/components/ProfileButton.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+import Button from '@material-ui/core/Button';
+
+import {withProfile} from "../providers/ProfileProvider";
+
+/**
+ * A button which allows sign in if the current user is anonymous or presents their username. Properties
+ * appropriate to the material-ui Button component (apart from ``component`` and ``href``) can be used.
+ *
+ * In addition to the basic component, ``ProfileButtonWithProfile`` is exported which is ``ProfileButton``
+ * already wired to the profile provider.
+ */
+const ProfileButton = ({ profile, ...otherProps }) => {
+  if(!profile) {
+    return (
+      <Button {...otherProps}>Sign in</Button>
+    );
+  }
+
+  if(profile.is_anonymous) {
+    return (
+      <Button component='a' href={profile.urls.login} {...otherProps}>
+        Sign in
+      </Button>
+    );
+  }
+
+  return (
+    <Button {...otherProps}>
+      { profile.username }
+    </Button>
+  );
+};
+
+const ProfileButtonWithProfile = withProfile(ProfileButton);
```

Comment:
I think this should be in a separate module, probably under ``src/containers``. From the [learn react blog](https://medium.com/@learnreact/container-components-c0e67432e005):

> A container does data fetching and then renders its corresponding sub-component. That’s it.

So there'd be a module, ``src/containers/ProfileButtonContainer.js`` with a correspondingly named component, ``ProfileButtonContainer`` which is simply ``ProfileButton`` wrapped with ``withProfile``.